### ISSUE TITLE
[ToggleGroup] aria-orientation not allowed on role=group element

### ISSUE THREAD

**Issue by hereismass** [mui/base-ui#4623]
# Bug report

## Current behavior

With the 1.4.0 update the ToggleGroup component violates the  WAI-ARIA 1.2 rule `aria-allowed-attr`.

`aria-orientation` is not a supported aria attribute for `role="group"` elements. (https://www.w3.org/TR/wai-aria-1.2/#group)

## Expected behavior

The simplest approach seems to use a different role. the `toolbar` role supports `aria-orientation`, but I'm not sure it matches really the ToggleGroup usage

## Reproducible example

base-ui doc: https://base-ui.com/react/components/toggle-group

## Base UI version

1.4.0

## Which browser are you using?

Chrome

## Which OS are you using?

Mac OS

## Additional context

We can override the role on the ToggleGroup so there is an easy workaround, but I guess it should be valid from the start :)
